src/bun.js/bindings/napi.cpp

```diff
@@ -2772,109 +2558,84 @@ extern "C" napi_status napi_create_bigint_words(napi_env env,
     size_t argc, const napi_value* argv,
     napi_value* result)
 {
-    NAPI_PREMABLE
-    if (UNLIKELY(!result)) {
-        return napi_invalid_arg;
-    }
-
-    Zig::GlobalObject* globalObject = toJS(env);
-    JSC::VM& vm = globalObject->vm();
-    auto throwScope = DECLARE_THROW_SCOPE(vm);
-
+    NAPI_PREAMBLE(env);
+    NAPI_CHECK_ARG(env, result);
+    NAPI_RETURN_EARLY_IF_FALSE(env, argc == 0 || argv, napi_invalid_arg);
     JSC::JSValue constructorValue = toJS(constructor);
-    if (UNLIKELY(!constructorValue.isObject())) {
-        return napi_function_expected;
-    }
-
+    NAPI_RETURN_EARLY_IF_FALSE(env, constructorValue.isObject(), napi_function_expected);
     JSC::JSObject* constructorObject = constructorValue.getObject();
     JSC::CallData constructData = getConstructData(constructorObject);
-    if (UNLIKELY(constructData.type == JSC::CallData::Type::None)) {
-        return napi_function_expected;
-    }
+    NAPI_RETURN_EARLY_IF_FALSE(env, constructData.type != JSC::CallData::Type::None, napi_function_expected);
+
+    Zig::GlobalObject* globalObject = toJS(env);
+    JSC::VM& vm = globalObject->vm();
 
     JSC::MarkedArgumentBuffer args;
-    args.fill(vm, argc, [&](auto* slot) {
-        memcpy(slot, argv, sizeof(JSC::JSValue) * argc);
+    args.fill(vm, argc, [&](JSValue* buffer) {
+        gcSafeMemcpy<JSValue>(buffer, reinterpret_cast<const JSValue*>(argv), sizeof(JSC::JSValue) * argc);
     });
 
     auto value = construct(globalObject, constructorObject, constructData, args);
-    RETURN_IF_EXCEPTION(throwScope, napi_pending_exception);
     *result = toNapi(value, globalObject);
-
-    RELEASE_AND_RETURN(throwScope, napi_ok);
+    NAPI_RETURN_SUCCESS_UNLESS_EXCEPTION(env);
 }
+
 extern "C" napi_status napi_call_function(napi_env env, napi_value recv_napi,
     napi_value func_napi, size_t argc,
     const napi_value* argv,
     napi_value* result_ptr)
 {
-    NAPI_PREMABLE
+    NAPI_PREAMBLE(env);
+    NAPI_RETURN_EARLY_IF_FALSE(env, argc == 0 || argv, napi_invalid_arg);
+    JSC::JSValue funcValue = toJS(func_napi);
+    NAPI_RETURN_EARLY_IF_FALSE(env, funcValue.isObject(), napi_function_expected);
+    JSC::CallData callData = getCallData(funcValue);
+    NAPI_RETURN_EARLY_IF_FALSE(env, callData.type != JSC::CallData::Type::None, napi_function_expected);
 
     Zig::GlobalObject* globalObject = toJS(env);
     JSC::VM& vm = globalObject->vm();
 
-    JSC::JSValue funcValue = toJS(func_napi);
-
-    if (UNLIKELY(!funcValue.isCell()))
-        return napi_function_expected;
-
-    JSC::CallData callData = getCallData(funcValue);
-    if (UNLIKELY(callData.type == JSC::CallData::Type::None))
-        return napi_function_expected;
-
     JSC::MarkedArgumentBuffer args;
-    if (argc > 0 && LIKELY(argv != nullptr)) {
-        auto end = argv + argc;
-        for (auto it = argv; it != end; ++it) {
-            args.append(toJS(*it));
-        }
-    }
+    args.fill(vm, argc, [&](JSValue* buffer) {
```

Comment:
`args.fill` does a slower thing that's only necessary when the size of the value is large enough to exceed the inline stack buffer limit

it will be slightly faster to do the for loop when the length is less than like 6

Reply:
(it adds itself to the GC, you can read the code in args.fill in webkit)